rfcs/NNNN-cdevents-provider/README.md

```diff
@@ -0,0 +1,140 @@
+<!--
+Add CDEvents to the list of available providers in Flux Notification controller. The user should be able to create mappings that tell it which CDEvents to send based on the flux event. It should then send that CDEvent to the specified URL.
+-->
+
+### User Stories
+
+Users of multiple CI/CD tools such as Tekton and Flux could use CDEvents as a way to enable interoperability.
+
+For example, a user may want a Tekton `pipeline` to run once a HelmRelease flux resource has succeeded in a Helm install. On successful helm install, Flux will emit an event with reason `InstallSucceeded` which the user will have mapped in their `Provider` configuration to correspond to an `Environment.Modified` CDEvent. The CDEvent `Provider` will then send a payload with that CDEvent, which will also contain data from the Flux event, to a CloudEvents broker that Tekton is subscribed to, and trigger a Pipeline Run within Tekton. 
+
+![User Stories Tekton](user-stories-provider.drawio.png)
+
+
+<!--
+Optional if existing discussions and/or issues are linked in the motivation section.
+-->
+
+### Alternatives
+
+There are currently no viable alternatives for the CDEvent Provider.
+<!--
+Certain use cases for CDEvents could be done alternatively using available providers such as the generic webhook.
+-->
+
+## Design Details
+
+Adding a Flux `Provider` for CDEvents that will send a CDEvent payload upon receiving a flux event from an alert. 
+
+The user will be able to define a Flux `Provider` custom resource and deploy it to their cluster. This provider configuration will allow the user to define a mapping of which Flux events correspond to which CDEvent to send. Once an alert is triggered for this provider, it will send the corresponding CDEvent, based on the Flux event that caused the alert. This CDEvent will be created using the [CDEvents Go SDK](https://github.com/cdevents/sdk-go). As well as the user-defined mapping, the provider will include some default mappings for Flux-Helm events in the case that the user does not provide a custom mapping.
+
+The CDEvents broker is not a part of this design and is left to the users to set up however they wish.
+
+Example Provider YAML:
+
+```yaml
+apiVersion: notification.toolkit.fluxcd.io/v1beta2
+kind: Provider
+metadata:
+  name: cdevents-provider
+  namespace: flux-system
+spec:
+  type: cdeventss
+  secretRef:
+    name: generic-token
+  address: http://10.100.0.17:5000
+  mapping: 
+    ReconciliationFinished: dev.cdevents.pipelinerun.finished
+    GitOperationFailed: dev.cdevents.incident.reported
```

Comment:
Hi sunny, apologies for the slow response, we have decided to in fact switch to an approach that focuses more on a mapping that is defined in the way you described rather than the custom mapping we had considered before. I will make the changes to the RFC proposal to reflect this.